### ISSUE TITLE
warning about incompatible rules with `--select ALL`

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [astral-sh/ruff#8470]
I'd rather not see warnings about incompatible rules with `--select ALL`. Do the folowing warnings about incompatible rules appear by default on purpose with `--select ALL`? 
* ```python
  Documentation.
  ```
* ```
  $ ruff --isolated --select ALL /path/to/file.py
  warning: `one-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible. Ignoring `one-blank-line-before-class`.
  warning: `multi-line-summary-first-line` (D212) and `multi-line-summary-second-line` (D213) are incompatible. Ignoring `multi-line-summary-second-line`.
  $ 
  ```
* ```
  $ ruff --version
  ruff 0.1.3
  ```

**Comment by zanieb** [comment]
If you don't want the warnings, I'd recommend explicitly ignoring two of those rules (whichever you do not want). In general, we do not recommend usage of `--ALL`.

**Comment by charliermarsh** [comment]
Yeah, these are intended, because the rules are legitimately in conflict. Ideally, we will resolve this when do recategorize the rules in the future. At present, the only other option I see is to straight-up omit those two specific rules (`D203` and `D213`) from `ALL`, which could also cause confusion in the other direction (though would be much more rare).

**Comment by charliermarsh** [comment]
(We could also consider removing those rules.)

**Comment by DimitriPapadopoulos** [comment]
It makes sense. However, newcomers who start with `ruff check --help` see:
```
$ ruff check -h
Run Ruff on the given files or directories (default)
[...]
Rule selection:
      --select <RULE_CODE>
          Comma-separated list of rule codes to enable (or ALL, to enable all rules)
      --ignore <RULE_CODE>
          Comma-separated list of rule codes to disable
[...]
```

**Comment by DimitriPapadopoulos** [comment]
They might be tempted to use option `--select ALL`. The result is that they see these warnings, which do make sense but might be disturbing for first time users.

**Comment by MichaReiser** [comment]
It could make sense to remove `ALL` from the CLI help. It isn't our recommended way of using ruff and is overused.

**Comment by zanieb** [comment]
@MichaReiser I think that may make sense after recategorization

**Comment by zanieb** [comment]
@MichaReiser I see the presence of sensible categories as essential to replace `ALL`. Until then, we'll just get more questions if we remove the hint.

**Comment by charliermarsh** [comment]
I don't think we should remove it from help, etc., until we have a reasonable alternative.

**Comment by charliermarsh** [comment]
I _do_ think we could consider removing these rules from `ALL`.

**Comment by zanieb** [comment]
I don't have a strong preference on removal from `ALL` but find excluding rules from `ALL` a weird pattern.

I'd rather close in favor of https://github.com/astral-sh/ruff/issues/1774

**Comment by charliermarsh** [comment]
It's weird, but so is the behavior here, and I do dislike that it's _so_ common to run into these warnings when the rules themselves are _so_ uncommon.

**Comment by tmeiczin** [comment]
We have over a million lines of code and currently use `ALL`  and disable about 40 rules (0.1.5 without preview). Without the `ALL` would we need to explicitly enable all 700 something rules? Enable all the categories, then specify individual ones we want disabled? That seems like it would get awkward? Or maybe I'm just selecting them wrong as it is?

I think many of the formatters and linters (maybe to a lesser extent) are going to be somewhat opinionated. I'm ok with that provided there is a reasonable amount of flexibility to alter the behavior to suit my projects. Maybe in addition to `ALL` add a `DEFAULT` which is a curated selection and chooses or disables rules with conflicts. Or possibly something like a `--select-conflict-defaults` which allows to specify which rules are enabled in case of conflicts.

**Comment by MichaReiser** [comment]
> We have over a million lines of code and currently use ALL and disable about 40 rules (0.1.5 without preview). Without the ALL would we need to explicitly enable all 700 something rules? Enable all the categories, then specify individual ones we want disabled? That seems like it would get awkward? Or maybe I'm just selecting them wrong as it is?

You would have to enable all categories that aren't enabled by default, which should only be a few (<10). The categories we're thinking about are similar to [clippy's](https://doc.rust-lang.org/clippy/) but with a few additional ones (e.g. `format` for formatting related lints)

**Comment by Avasam** [comment]
Without enabling `ALL`, these still come in conflict if you enable the `D` category. I'd rather fully opposite rules like these be configurable instead. (combine the codes and use a config option to decide on the behaviour).

That would also make more sense in the context of shared configs https://github.com/astral-sh/ruff/issues/12352

**Comment by nm-remarkable** [comment]
> If you don't want the warnings, I'd recommend explicitly ignoring two of those rules (whichever you do not want). In general, we do not recommend usage of `--select ALL`.

In a codebase I work on we have this configuration (partial), however I still get the warning even when I ignore all documentation:

```toml
[tool.ruff]
line-length = 120

# Supported rules https://github.com/charliermarsh/ruff/#supported-rules
lint.select = ["ALL"]

lint.ignore = [
    "ANN",    # flake8-annotations: Allow untyped code
    "E501",   # Line lenght higher than 88 characters
    "D",      # pydocstyle: Allow undocummented code
```

```log
[nuno@fedora xochitl]$ uvx ruff check .
warning: `multi-line-summary-first-line` (D212) and `multi-line-summary-second-line` (D213) are incompatible. Ignoring `multi-line-summary-second-line`.
All checks passed!
```

**Comment by MichaReiser** [comment]
@nm-remarkable 

I just tried 

```
[tool.ruff]
line-length = 120

# Supported rules https://github.com/charliermarsh/ruff/#supported-rules
lint.select = ["ALL"]

lint.ignore = [
    "ANN",    # flake8-annotations: Allow untyped code
    "E501",   # Line lenght higher than 88 characters
    "D",      # pydocstyle: Allow undocummented code
]

lint.pydocstyle.convention = "google"
```

(with or without the `pydocstyle.convention` line) and I'm unable to reproduce. Could you open a new issue and provide more details (full ruff configuration, what ruff version are you using)

**Comment by czechnology** [comment]
I wish that Ruff would include some default all-covering family of rules without the conflicting rules.
I do like to take advantage of all the advice that the rules give me and only explicitly disable the ones which don't work for me.

However, right now if I `select = ["ALL"]`, I need to exclude quite many individual conflicting rules and repeat this in all our projects.

Please add something like `select = ["ALL_COMPAT"]` or `select = ["STRICT"]` to easily enable all rules without any conflicts. If there is something to pick (like `D203` _xor_ `D211`, or `D212` _xor_ `D213`), just pick one! Let us have a default default to follow.

**Comment by lypwig** [comment]
> If there is something to pick (like D203 xor D211, or D212 xor D213), just pick one! Let us have a default default to follow.

I totally support this.
I usually use `select = ["ALL"]` and then ignore some rules, and those warnings are annoying.
I know I can disable them by ignoring D203 and D212 rules, but it adds mental load as every time I see them in my config I ask to myself why I wanted them in the first place.
